kin-sdk/kin-sdk-lib/src/test/kotlin/kin/sdk/internal/queue/PaymentQueueManagerTest.kt

```diff
@@ -0,0 +1,170 @@
+        verify(queueScheduler).removePendingTask(any())
+        verify(queueScheduler).removeAllPendingTasks()
+
+        verify(txTaskQueueManager).enqueue(pendingPayments)
+    }
+
+    @Test
+    fun `enqueue, then wait for queue timeout, after the timeout the list will be empty`() {
+        //given
+        val pendingPayment1 = PendingPaymentImpl(destinationAccount, sourceAccount, amount)
+        val pendingPayment2 = PendingPaymentImpl(destinationAccount, sourceAccount, amount)
+        val pendingPayment3 = PendingPaymentImpl(destinationAccount, sourceAccount, amount)
+
+        //when
+        paymentQueueManager.enqueue(pendingPayment1)
+        Thread.sleep(Constants.DELAY_BETWEEN_PAYMENTS_MILLIS - Constants.ONE_MILLI)
+        paymentQueueManager.enqueue(pendingPayment2)
+        Thread.sleep(Constants.DELAY_BETWEEN_PAYMENTS_MILLIS - Constants.ONE_MILLI)
+        paymentQueueManager.enqueue(pendingPayment3)
+
+        //then
+        assertThat(paymentQueueManager.pendingPaymentCount, equalTo(3))
+        Thread.sleep(Constants.DELAY_BETWEEN_PAYMENTS_MILLIS)
+        assertThat(paymentQueueManager.pendingPaymentCount, equalTo(0))
+
+        val pendingPayments: MutableList<PendingPayment> = mutableListOf()
+        pendingPayments.add(pendingPayment1)
+        pendingPayments.add(pendingPayment2)
+        pendingPayments.add(pendingPayment3)
+
+        verify(queueScheduler).removeAllPendingTasks()
+        verify(txTaskQueueManager).enqueue(pendingPayments)
+    }
+
+    @Test
+    fun `enqueue max items, after the queue has been filled then the list will be empty`() {
+        //given
+        val pendingPayment1 = PendingPaymentImpl(destinationAccount, sourceAccount, amount)
+        val pendingPayment2 = PendingPaymentImpl(destinationAccount, sourceAccount, amount)
+        val pendingPayment3 = PendingPaymentImpl(destinationAccount, sourceAccount, amount)
+        val pendingPayment4 = PendingPaymentImpl(destinationAccount, sourceAccount, amount)
+        val pendingPayment5 = PendingPaymentImpl(destinationAccount, sourceAccount, amount)
+
+        //when
+        paymentQueueManager.enqueue(pendingPayment1)
+        paymentQueueManager.enqueue(pendingPayment2)
+        paymentQueueManager.enqueue(pendingPayment3)
+        paymentQueueManager.enqueue(pendingPayment4)
+        assertThat(paymentQueueManager.pendingPaymentCount, equalTo(4))
+        paymentQueueManager.enqueue(pendingPayment5)
+
+        //then
+        assertThat(paymentQueueManager.pendingPaymentCount, equalTo(0))
+
+        val pendingPayments: MutableList<PendingPayment> = mutableListOf()
+        pendingPayments.add(pendingPayment1)
+        pendingPayments.add(pendingPayment2)
+        pendingPayments.add(pendingPayment3)
+        pendingPayments.add(pendingPayment4)
+        pendingPayments.add(pendingPayment5)
+
+        verify(queueScheduler).removeAllPendingTasks()
+        verify(txTaskQueueManager).enqueue(pendingPayments)
+    }
+
+    @Test
```

Comment:
Worth waiting the timeout and check there its empty

Reply:
I have a specific test for that, called:
`enqueue, then wait for queue timeout, after the timeout the list will be empty`